### DISCUSSION TITLE
Better offramping docs for new hunters

### DISCUSSION THREAD

**Discussion by Keyrxng** [ubiquity/work.ubq.fi#53] [Q&A]
I'll likely be left for dust shortly now that new hunters who seem like they'll give me a good run for my money are starting to drip into the devpool, I love to see it.

With that said, I came in knowing my way around Web3 having been in it for a year previous, so I was off to a good start. Jordan has cleared a couple of tasks and attempted to move his funds but unfortunately has lost that payout due to user error. He contacted me asking for assistance, and I gave as much as I could in his situation.

Every one of us has done it at one point or another, and lost funds due to a bad transfer/misunderstanding, no shade on Jordan at all, it's commonplace for those new to Web3.

<hr/>

This issue aims to set up a document/reference point that covers team-trusted/contributor-trusted methods in moving their funds from Permit > Fiat.

Core team methods would likely carry more weight than say my own methods, but I understand everyone may not be as comfortable detailing each step they take to off-ramp exactly, however a structured document with paths to follow and common problem solutions is 100% required. 

Jordan took it like a champ and put it down to user error, but this should not be an event that is easily repeatable for a new hunter. In this case, an attempt was made to bridge the assets cross-chain without actually using a bridge and so the funds were delivered to an inaccessible wallet on the same chain.

<hr/>

RFC @0x4007 @gentlementlegen @EresDev @sergfeldman @molecula451 @rndquu @devpanther

_original context https://github.com/ubiquity/pay.ubq.fi/issues/217#issuecomment-2066259371_

**Comment by Keyrxng** [comment]
I think starting bottom up is best. We know permits are all on Gnosis these days and it's cheaper so, start by advising them to 

A) Find a CEX in there country that supports Gnosis chain deposits (easiest offramp, hard to find gnosis supporting CEX at least in my exp in UK)
B) If A cannot be done, find a CEX that supports Polygon. (most CEX support it and still v cheap)
C) If B cannot be done, use Ethereum....

If A: walk through how to transfer, best practices etc. (for instance, I copy and paste the CEX address and send it to the Telegram chat on TG mobile. I then easily copy it on TG desktop and paste it into MM wallet from there)

If B: The user needs to bridge, walk them through bridging in general, then specifically the methods the team/contributors trust and use to off-ramp. This should also include all advice from `If A:` as this will require ~3 separate transfers. The user should be aware they'll need to send funds from the CEX to the wallet on the destination bridge chain in order to move the asset to the CEX after bridging unless bridging directly to the CEX address (something that I would not do personally but imagine others have no issues with it)

<hr/>

I'm not great at docs etc but I'm more than happy to give my input on paths, solutions, best practices, tips and tricks whatever. Unsure whether to do so and just post it as a comment here, collaborate with someone on it or what but I'm happy to get involved.

**Comment by molecula451** [comment]
You know this thing changes and evolves, if we concrete something like a way or mechanism that someone do might no be fit for other contributor, that's why many exists, there's currently not CEX that support Gnosis as far as I can tell, for bridges there are many known bridges that ramp to other chains, or just the Gnosis official to ramp back to Ethereum-Dai

**Comment by Keyrxng** [comment]
I hear you it'll likely be outdated in a few months, but it would be better to have something at least the foundational stuff that won't change like the act of bridging, transferring, the dos and don'ts etc

As for paths to follow, if the doc states `at time of writing the most common methods used by the team and active contributors for offramping are...if still you are unsure reach out on telegram before making any transfers` that should be disclaimer enough

> mechanism that someone do might no be fit for other contributor,

That's why I tried to rally the troops, we seem to be pretty diverse in terms of locations (I think) and so I assumed the same with methods of off-ramping and I'm thinking that something is better than nothing for a person who is completely new/long hiatus from the scene

> there's currently not CEX that support Gnosis 

I thought I was just my luck

**Comment by 0x4007** [comment]
From some direct messages:

![image](https://github.com/ubiquity/work.ubq.fi/assets/4975670/a772ea7f-d730-4be7-a24b-9022078c892f)

1. Unwrap your WXDAI into xDAI using any major DEX. This will handle the method call for you. You can use [CowSwap](https://swap.cow.fi/#/100/swap/WXDAI/xDAI).

2. Go to the [Gnosis Chain Bridge](https://bridge.gnosischain.com/) and manually enter an amount that is not 100%. Leave about a dollar’s worth of xDAI in your wallet for transactions.

3. Complete the bridge transaction and wait for 20 blocks (approximately 4.3 minutes).

4. Check your [transaction history](https://bridge.gnosischain.com/history). When available, click the claim button to withdraw to mainnet.

5. Swap DAI to USDC and transfer to Coinbase for cash out using [CowSwap](https://swap.cow.fi/#/1/swap/DAI/USDC). Use the gear icon on the top right to specify the swap beneficiary for a direct deposit to your Coinbase address.

---

Some caveats:

- Transaction fees on Ethereum mainnet can be high during busy periods. Be aware of gas fees when making transactions. I use the [DeFi Saver Gas Prices Chrome extension](https://chromewebstore.google.com/detail/defi-saver-gas-prices-ext/afgfdkloegmghldbalmenklokhlifphe).

- Another useful tool for Ethereum network activity is [Etherscan](https://etherscan.io/gastracker).

---

Additional advice:

- Jin explored bridging to another network for cash out. Using a different bridge, you can transfer from Gnosis Chain to Base (network) and then to Coinbase. Coinbase supports deposits from Base, and gas fees on Base are much lower than on Ethereum mainnet. Jin uses the [Bungee Exchange](https://www.bungee.exchange/) for this process, though I haven't tried it myself yet.

**Comment by Keyrxng** [comment]
# Setup and Bridging to Polygon

The bridge I use is [Connext Network Bridge](https://bridge.connext.network/DAI-from-gnosis-to-polygon?symbol=WXDAI) and they support a fair few chains.

If you need to move your WXDAI onto the Polygon chain (or any chain the bridge and your CEX supports) to deposit it into your CEX account to "cash out", follow these steps:

## Initial Setup

1. **Send Matic**: Send $5 of Matic (or native chain token, ETH, BNB, ...) to your wallet address on the Polygon chain from your CEX account. This will be the gas fees used to send your permit rewards into the CEX to be able to cash out.
2. **Send DAI**: Send $5 of DAI to the same wallet address on the Polygon chain also. This is the gas fees to claim your permit.
3. **Go to Bridge**: Select the chain you are bridging from and to (Polygon > Gnosis) and enter $4 worth of DAI then bridge. I say $4 because paying gas on the destination chain may cost upfront on the sending chain.
4. **Permit Now Claimable**: Now your Gnosis address has $4 worth of XDAI as does your Polygon address in Matic. Claim the permit to receive your WXDAI reward.

![image](https://github.com/ubiquity/work.ubq.fi/assets/106303466/173f3fcf-a8a8-4a85-b24d-d8b0be3faf8c)


## Post-Claim Process
1. **Bridge**: Go back to the bridge and swap the chains around (Gnosis > Polygon), select WXDAI and enter the amount you want to bridge and wait until confirmed (UI has a retro car driving into the sunset that keeps you informed). 
2. **Switch Chain**: Change the chain inside your wallet to Polygon which now has DAI in it automatically unwrapped from your WXDAI reward.
3. **Transfer to CEX**: Transfer the DAI to your CEX account using your wallet interface.

![image](https://github.com/ubiquity/work.ubq.fi/assets/106303466/6bf7355b-55c2-411b-9514-40e6624e96bc)

# Tips and Best Practices

- **Account**: If you own the private key to account A on EVM chain A, you own that same address on all EVM chains. Only bridge/transfer to an address you are confident that you own the private keys to otherwise you will lose your funds. Each CEX operates differently but must do not help in retrieving funds sent to an incorrect address/incorrect destination chain, so when sending to your CEX ensure that the address, chain (and potentially memo) info is correct or you will lose your funds.

- **Delays**: Typically < 3m but can see multiple hour delays rarely if the protocol is congested, all txs are guaranteed so nothing to fear. 

- **Save the CEX Address**: If using MetaMask, Open settings (three vertical dots), open contacts, select add contact. Do this once and you don't have to worry about it again. I will say that some CEX have been known to update their deposit addresses so every once in a while double check on the CEX side that the address is still valid.

- **Entering Addresses**: Never manually enter an address. I copy the address from the CEX app and open Telegram, then open the private chat with the actual Telegram account. I then paste the address on mobile and copy it on desktop. This way, I'm 100% sure the address is correct and avoid any mistakes.

- **Gas Fees**: Because fees are so cheap on Gnosis and Polygon you only need to worry about having a little Matic and XDAI in your wallet on the respective chains. A typical tx costs < $0.05 on these chains so even $1 worth of Matic and XDAI will last you a long time.

- **Avoid mobile**: I avoid doing any of this on mobile because it's easier to make mistakes, as I've lost funds in the past myself

- **Add to Wallet**: You may have to import DAI and/or WXDAI into your wallet in order to interact with it (such is the case with metamask). Google the token + coinmarketcap or coingecko and use the little metamask widget to add it or copy paste the address manually inside the wallet selecting option `+import tokens`

- **Direct to CEX**: I do not do this but it can be done. Change the default recipient address in the bridge settings to that of your CEX deposit address which removes the need to keep your Polygon address funded (after the initial funding of your Gnosis wallet of course) and cuts a step out of my process. It's easier to debug if something goes wrong with the additional step, but a little less convenient.

![image](https://github.com/ubiquity/work.ubq.fi/assets/106303466/4d2ecf2e-4285-42c7-b3c6-d207e3a39e12)

**Comment by EresDev** [comment]
I have used [jumper.exchange](https://jumper.exchange/) to move wxdai/xdai gnosischain to USDT on the BNB chain. 

Jumper exchange is also listed on the gnosischain bridge [website](https://bridge.gnosischain.com/). 

It offers good rates, low fees, and fast transfer. 

The process is very simple:

- go to https://jumper.exchange/
- connect your web3 wallet, e.g. metamask.
- select your "From" chain and token, which is gnosischain and xDai/wxDai
- select "To" chain and token. Right now it supports  about 20+ chains and almost all major tokens. 
- Insert amount, and it shows you all possible routs. Here you can select the best route that suits you depending on fee and time to process. 


![image](https://github.com/ubiquity/work.ubq.fi/assets/11886219/891916c3-665f-4f29-bc2f-5c7036f26a2a)

**Comment by molecula451** [comment]
jumper it's more like a bridge aggregator where it finds you the best route available along with the possible bridge provider

**Comment by rndquu** [comment]
> methods in moving their funds from Permit > Fiat

This is pretty much county/region specific so we won’t cover all possible ways for off-ramping.

Overall this issue already contains solid advices for off-ramping so it makes sense to convert this issue to a github discussion and reference it in case somebody has issues with transferring WXDAI from gnosis.

**Comment by gentlementlegen** [comment]
To throw my 2 cents here, I usually use https://www.bungee.exchange/ which works great with low fees, automatic refuel, and a very responsive help desk as well. You can convert directly from WXDAI to USDC that can easily be cashed out.

<img width="550" alt="image" src="https://github.com/ubiquity/work.ubq.fi/assets/9807008/f4b3c143-50a4-46a3-883d-df6658072b3e">

I agree that this conversation should be pinned somewhere for future reference if anyone needs it.
